### PR TITLE
Switch to `entry_points().select` to get rid of DeprecationWarning

### PR DESCRIPTION
**Pull request by theXYZT**
The `entry_points` API now recommends using the `.select` method. Using `.get` gives a DeprecationWarning.

### CODE REVIEW

**Comment by theXYZT** [comment]
I tried adding a Python 3.10 test with coverage but that didn't change codecov's coverage values at all :(

**Comment by mhvk** [comment]
The usual trick is just to mark the lines that are not covered with `# pragma: no cover`. Though if you make this a python version check, it /may/ be smart enough to realize it cannot work (really not sure).

Aside, somewhat surprised by the API change; not quite obvious why.

File: baseband/io/__init__.py [hunk 1]
```diff
@@ -61,8 +61,14 @@ def __getattr__(attr):
 
         # Note: again do not presume the entry points exist, since we may
         # be in a pure source checkout.
+
+        if hasattr(entry_points(), 'select'):
+            selected_entry_points = entry_points().select(group='baseband.io')
```
Comment (mhvk) [inline baseband/io/__init__.py:66]: Might as well use `entry_points(group='baseband.io')`, which seems nicer (though apparently the two are exactly equivalent)

Comment (theXYZT) [reply baseband/io/__init__.py]: Made changes and swapped to try/except also!

**Comment by theXYZT** [comment]
Woo! I figured out how the coverage was working and fixed it. Turns out codecov automatically merges reports, so you just have to make sure the `-cov` suffix on tox environments is used for the different python versions.

**Comment by theXYZT** [comment]
The remaining checks will not pass because @mhvk comments out the docs build workflow and I updated the Windows check to Python 3.10 so I don't know why the 3.8 check still exists (probably has to do with repository settings)

**Comment by mhvk** [comment]
Not sure it is worth running coverage on multiple streams - that costs substantial extra cycles for rather little gain.

Not sure what is up with the docs build...

**Comment by theXYZT** [comment]
Would you be willing to accept and merge the PR and then change the github workflow to fit your preferences?

**Comment by mhvk** [comment]
Yes, in principle, but I'm a bit worried that the hung checks mean that all builds will fail if I merge. But in any case, happy to let that be *my* worry! I can just fiddle and edit your PR.

**Comment by mhvk** [comment]
Thanks, NIkhil!

**Pull request merged**